src/LEGO.AsyncAPI.Readers/V2/AsyncApiV2VersionService.cs

```diff
@@ -67,9 +67,21 @@ public AsyncApiV2VersionService(AsyncApiDiagnostic diagnostic)
                             Id = reference,
                         };
                     }
+
+                    var asyncApiReference = new AsyncApiReference();
+                    if (reference.StartsWith("/"))
```

Comment:
Probably worth noting that I believe the spec also supports "relative" paths. So they could conceivably start with `.` and `..` as well.

Reply:
This is already fully supported, gotta have to check for the case of having a fragment though.

Something like 
`./myjsonfile.json/fragment`

that would simply fall under 'externalresource' and not `isFragment: true`

Reply:
So, just to clarify, something like a schema stored as a yaml file separately using cloudevents or some other schema would be an `externalResource` and *also* a fragment. But a reference to an entire AsyncAPI doc would simply be an `externalResource`?

Reply:
I think fragments is, when a subset is referenced inside another Json file. 
So `jsonfile.json#/someLocation`
But I got dizzy from all of the RFCs involved. 
I need to be determine 100% what "fragment" actually refers to.

Reply:
After some investigating, ive come to the understanding that there are 2 types of references
1. Fragments 
`./external-doc.json#/fragment` different file document. (relative path)
`/fragments/myFragment` same document.
`https://example.com/document.json#/fragment` different external file document.
1. Internal 
`#/components/schemas/Pet` (we already support those)
1. 'Full' document Referenes
`./external-doc.json` We can choose to support this (although it technically does not align with the spec.')

and is a mix of 
[Json Reference](https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03)
[JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901)
[Uri](https://datatracker.ietf.org/doc/html/rfc3986)

The main point is paragraph 3 and 4 of the Json Reference RFC

> [3](https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03#section-3).  Syntax
> 
>    A JSON Reference is a JSON object, which contains a member named
>    "$ref", which has a JSON string value.  Example:
> 
>    { "$ref": "http://example.com/example.json#/foo/bar" }
> 
>    If a JSON value does not have these characteristics, then it SHOULD
>    NOT be interpreted as a JSON Reference.
> 
>    The "$ref" string value contains a URI [[RFC3986](https://datatracker.ietf.org/doc/html/rfc3986)], which identifies
>    the location of the JSON value being referenced.  It is an error
>    condition if the string value does not conform to URI syntax rules.
>    Any members other than "$ref" in a JSON Reference object SHALL be
>    ignored.
> 

> 
> [4](https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03#section-4).  Resolution
> 
>    Resolution of a JSON Reference object SHOULD yield the referenced
>    JSON value.  Implementations MAY choose to replace the reference with
>    the referenced value.
> 
>    If the URI contained in the JSON Reference value is a relative URI,
>    then the base URI resolution MUST be calculated according to
>    [[RFC3986], section 5.2](https://datatracker.ietf.org/doc/html/rfc3986#section-5.2).  Resolution is performed relative to the
>    referring document.
> 
>    If a URI contains a fragment identifier, then the fragment should be
>    resolved per the fragment resolution mechansim of the referrant
>    document.  If the representation of the referrant document is JSON,
>    then the fragment identifier SHOULD be interpreted as a
>    [[JSON-Pointer](https://datatracker.ietf.org/doc/html/draft-pbryan-zyp-json-ref-03#ref-JSON-Pointer)].
>

Reply:
Ive pushed tests that should verify this behaviour

Reply:
Ok, interesting, definitely adds some clarity.

Especially so that "'Full' document References" aren't in this RFC because I've seen them used a lot on AsyncAPI slack to allow sharing of resources between OpenAPI schemas, AsyncAPI schemas and schema registry. 

Is it significant that all the examples are to .json files? Or can we allow for .yaml as well?

Similar context in terms of spec intention vs facilitating actual usage: I'm never sure with this stuff because I'm sure there's a similar situation with `servers` where the AsyncAPI spec doesn't actually specify its contents can be `$ref`'d but I've seen a many members of the core AsyncAPI team themselves reffing the servers object in their config.

Reply:
Given the `IReferenceLoader` pattern on my PR we could actually just push this concern to the user. They can decide how they want to read when using external files. We could provide them with implementations for the canonical uses outlined. But if they want to do something extremely weird like store their references then they've got the interface to implement around their needs.

Reply:
My thoughts exactly